logstreams/src/test/java/io/camunda/zeebe/logstreams/log/LogStreamWriterTest.java

```diff
@@ -277,15 +241,6 @@ public void shouldWriteNullKeyByDefault() {
     assertThat(getWrittenEvent(position).getKey()).isEqualTo(LogEntryDescriptor.KEY_NULL_VALUE);
   }
 
-  @Test
-  public void shouldFailToWriteEventWithoutValue() {
-    // when
-    final long pos = tryWrite(appendEntry.reset());
```

Comment:
:question: Is this not possible anymore? I guess this would still be part of the contract? Though I'm all for removing it from the contract in general

Reply:
Considering LogAppendEntry is an interface (so can't guarantee it will always be non-null), I guess we need to do it in `tryWrite`.

:thought_balloon: I guess if we start having more validation logic in try write, then we can revisit the concept of splitting writer out of sequencer :shrug:

Reply:
Added some basic validation: [a592140](https://github.com/camunda/zeebe/pull/11165/commits/a592140c0e2116de6f6d670b1a786069cb11badf)